avr-hal-generic/src/wdt.rs

```diff
@@ -55,6 +61,21 @@ pub trait WdtOps<H> {
     fn raw_stop(&mut self);
 }
 
+#[derive(Copy, Clone)]
+/// The mode that dictates how the watchdog should behave.
+/// This needs to be called **after** the `start` function.
+pub enum WdtMode {
+    /// Trigger the interrupt `WDT`, but do not perform a softreset.
+    /// **Warning**: On many `atmega` devices, a watchdog timeout resets the mode automatically, to
+    /// prevent the microcontroller from getting stuck in a infinite loop. So subsequent timeouts
```

Comment:
Please be specific here: Which MCUs actually behave like this and which don't?  Is it even a good idea to distingush about this or should we rather require the call of `rearm()` on _every_ AVR microcontroller?